fuzzer/src/main.rs

```diff
@@ -0,0 +1,223 @@
+    services::api::contract_class::{ContractClass, EntryPointType},
+    utils::{calculate_sn_keccak, Address},
+};
+
+use std::{
+    collections::{HashMap, HashSet},
+    path::PathBuf,
+};
+
+use std::fs;
+use std::process::Command;
+
+
+fn main() {
+    println!("Starting fuzzer");
+    let mut iteration = 0;
+    fs::create_dir("cairo_programs");
+    loop {
+        fuzz!(|data: &[u8]| {
+
+        iteration += 1;
+
+        // ---------------------------------------------------------
+        //  Create the content of the .cairo file with a function that writes the random data in the counter
+        // ---------------------------------------------------------
+        let file_content1 = "
+        %lang starknet
+        from starkware.cairo.common.cairo_builtins import HashBuiltin
+        
+        @storage_var
+        func _counter() -> (res: felt) {
+        }
+        
+        @external
+        func write_and_read{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res:felt) {
+             _counter.write('";
+
+        let input = data_to_ascii(data);
+        
+        let file_content2 = "');
+            return _counter.read();
+        }
+        ";
+        let file_name = format!("cairo_programs/output-{}.cairo", iteration);
+
+        let file_content  = file_content1.to_owned() + &input + file_content2 ;
+
+        println!("{:?}", file_content);
+        // ---------------------------------------------------------
+        //  Create the .cairo file 
+        // ---------------------------------------------------------
+        fs::write(file_name, file_content.as_bytes());
+
+        // ---------------------------------------------------------
+        //  Compile the .cairo file to create the .json file
+        // ---------------------------------------------------------
+
+        let cairo_file_name = format!("cairo_programs/output-{}.cairo", iteration);
+        let json_file_name = format!("cairo_programs/output-{}.json", iteration);
+
+        let _output = Command::new("starknet-compile")
+                    .arg(cairo_file_name.clone())
+                    .arg("--output")
+                    .arg(json_file_name.clone())
+                    .output()
+                    .expect("failed to execute process");
+
+        // ---------------------------------------------------------
+        //  Create program and entry point types for contract class
+        // ---------------------------------------------------------
+
+        let path = PathBuf::from(json_file_name.clone());
+        let contract_class = ContractClass::try_from(path).unwrap();
+        let entry_points_by_type = contract_class.entry_points_by_type().clone();
+
+        let storage_entrypoint_selector = entry_points_by_type
+            .get(&EntryPointType::External)
+            .unwrap()
+            .get(0)
+            .unwrap()
+            .selector()
+            .clone();
```

Comment:
You don't need `.clone()` there:
```suggestion
        let storage_entrypoint_selector = contract_class
            .entry_points_by_type()
            .get(&EntryPointType::External)
            .unwrap()
            .get(0)
            .unwrap()
            .selector()
            .clone();
```

Reply:
done